l10n_it_delivery_note/migrations/14.0.2.6.0/pre-migrate.py

```diff
@@ -0,0 +1,47 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    # Get every carrier that was a 'res.partner' and create a 'delivery.carrier'
+    env.cr.execute(
+        "select id, carrier_id from stock_delivery_note where carrier_id IS NOT NULL"
+    )
+    res = env.cr.fetchall()
+
+    for (delivery_note_id, carrier_id) in res:
+        partner_carrier_id = env["res.partner"].browse(carrier_id)
+
+        carrier_product = env["product.product"].create(
+            {
+                "name": partner_carrier_id.name,
+                "type": "service",
+                "sale_ok": False,
+                "purchase_ok": False,
+                "list_price": 0.0,
+                "invoice_policy": "order",
+            }
+        )
+
+        delivery_carrier_id = env["delivery.carrier"].create(
```

Comment:
unica osservazione tu fai sempre la create del `delivery.carrier` in questo caso avrai vari delivery carrier con la stesso nome perché non c'è un raggruppo per partner quindi anche se il partner sul secondo DdT è lo stesso crea di nuovo il `delivery.carrier`

Reply:
Grazie per avermelo fatto notare, mi era totalmente sfuggito.

Ora funziona come dovrebbe, fammi sapere se l'implementazione ti piace :)